File: src/app/models/game.model.ts
import {User} from './user.model';

interface IGame {
  GameId: string;
  Name: string;
  Owner: User;
  MinPlayers: number;
  MaxPlayers?: number;
  MyRating: number;
  AvgRating?: number;
  DateCreated: Date;
}

export class Game {

  public GameId: string;
  public Name: string;
  public Owner: User;
  public MinPlayers: number;
  public MaxPlayers?: number;
  public MyRating: number;
  public AvgRating?: number;
  public DateCreated: Date;

  constructor(obj?: IGame) {
    this.GameId = obj && obj.GameId || "";
    this.Name = obj && obj.Name || "";
    this.Owner = obj && obj.Owner || new User();
    this.MinPlayers = obj && obj.MinPlayers || null;
    this.MaxPlayers = obj && obj.MaxPlayers || null;
    this.MyRating = obj && obj.MyRating || 0;
    this.AvgRating = obj && obj.AvgRating || null;    
    this.DateCreated = obj && obj.DateCreated || new Date();
    
  }
}

File: GameNightBuddy-Server/src/GameNightBuddy-Server/Models/Game.Model.cs
using GameNightBuddy_Server.ViewModels;
using System;
using System.Collections.Generic;

namespace GameNightBuddy_Server.Models
{
  public class Game
  {
    // Primary Key
    public Guid GameId { get; set; }

    // Foreign Key
    public Guid UserId { get; set; }
    public User User { get; set; }
    // Foreign Keys Elsewhere
    public List<GameNightGame> GameNightGames { get; set; }
    public List<Match> Matches { get; set; }
    public List<GameRating> GameRatings { get; set; }

    // Unique Columns
    public string Name { get; set; }
    public int MinPlayers { get; set; }
    public int MaxPlayers { get; set; }

    // Default
    public DateTime DateCreated { get; set; }
    public string UserCreated { get; set; }
    public DateTime DateUpdated { get; set; }
    public string UserUpdated { get; set; }
    public Boolean IsActive { get; set; }

    public Game() { }

    public Game(GameViewModel vm)
    {
      GameId = vm.GameId.Length > 0 ? new Guid(vm.GameId) : new Guid();
      UserId = vm.Owner.UserId;
      Name = vm.Name;
      MinPlayers = vm.MinPlayers;
      MaxPlayers = vm.MaxPlayers;
    }
  }
}
